R/Connection.R

```diff
@@ -608,24 +608,94 @@ setMethod(
+#' Driver names are first processed with `keep`, then `filter`. Thus, if a
+#' driver name is in both `keep` and `filter`, it won't appear in output.
+#'
+#' If a driver has multiple attributes, there will be one row per attribute,
+#' each with the same driver `name`. If a given driver name does not have any
+#' attributes, the function will return one row with the driver `name`, but
+#' the last two columns will be `NA`.
+#'
+#' @section Configuration:
+#'
+#' This function interfaces with the driver manager to collect information
+#' about the available driver names.
+#'
+#' For **MacOS and Linux**, the odbc package supports the unixODBC driver
+#' manager. unixODBC looks to the `odbcinst.ini` _configuration file_ for
+#' information on driver names. Install unixODBC with `brew install unixodbc`,
+#' and find the location(s) of your `odbcinst.ini` file(s) with `odbcinst -j`.
+#'
+#' In this example `odbcinst.ini` file:
+#'
+#' ```
+#' [MySQL Driver]
+#' Driver=/opt/homebrew/Cellar/mysql/8.2.0_1/lib/libmysqlclient.dylib
+#' ```
+#'
+#' ...the driver name is `MySQL Driver`, which will appear in the `name`
+#' column of this function's output. To pass the driver name as the `driver`
+#' argument to [dbConnect()], pass it as a string, like `"MySQL Driver"`.
+#'
+#' **Windows** is [bundled](https://learn.microsoft.com/en-us/sql/odbc/admin/odbc-data-source-administrator)
+#' with an ODBC driver manager.
+#'
+#' In this example, function output would include 1 row: the `name` column
+#' would read `"MySQL Driver"`, `attribute` would be `"Driver"`, and `value`
+#' would give the file path to the driver. Additional key-value pairs
+#' under the driver name would add additional rows with the same `name` entry.
+#'
+#' When a driver is configured with a driver manager, information on the driver
+#' will be automatically passed on to [dbConnect()] when its `driver` argument
+#' is set. Further, if a data source name (DSN) is configured with a driver,
+#' information on the driver will automatically passed on to to [dbConnect()]
+#' without needing to pass the `driver` argument. For an example, see the same
+#' section in the [odbcListDataSources()] help-file.
+#'
+#' @inheritSection odbcListDataSources Learn more
+#'
+#' @seealso
+#' [odbcListDataSources()] collects information on the available data source
+#' names (DSNs). On MacOS and Linux, the analogous configuration file to
+#' `odbcinst.ini` for DSNs is `odbc.ini`.
+#'
+#' @examplesIf FALSE
+#' odbcListDrivers()
+#'
+#' # retain only attributes on the SQLite Driver
+#' odbcListDrivers(keep = "SQLite Driver")
+#'
+#' # remove results pertaining to the SQLite Driver
+#' odbcListDrivers(filter = "SQLite Driver")
+#'
+#' # to always keep or remove only results for a given driver,
+#' # use the odbc.drivers_keep and odbc.drivers_filter options
+#' options(odbc.drivers_keep = "SQLite Driver")
+#' odbcListDrivers()
+#'
+#' # to set that option back to its default:
+#' options(odbc.drivers_keep = NULL)
```

Comment:
IMO `keep` and `filter` are sufficiently specialised that they don't need examples.

Reply:
[c99a80b](https://github.com/r-dbi/odbc/pull/649/commits/c99a80b2dd361ccecc5bbce11c63be7fc5f3e514).